### PR TITLE
ROC-3702: Check and send and confirmation changes

### PR DESCRIPTION
**Pull request by dharmendrak**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-2703

### Change description
Add heck and send and Confirmation for part admission journey

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/main/features/response/draft/responseDraft.ts [hunk 1]
```diff
@@ -141,11 +153,18 @@ export class ResponseDraft extends DraftDocument {
   // is incomplete. ROC-3658 should revisit once 'statement of means' flow is complete.
   public isResponseFullyAdmittedWithInstalments (): boolean {
     return this.isResponseFullyAdmitted()
-      && this.fullAdmission
-      && this.fullAdmission.paymentOption
+      && this.fullAdmission !== undefined
+      && this.fullAdmission.paymentOption !== undefined
       && (this.fullAdmission.paymentOption.option === DefendantPaymentType.INSTALMENTS)
```
Comment (yarpo) [inline src/main/features/response/draft/responseDraft.ts]: can you please either add brackets to conditions above or remove this one?

File: src/main/features/response/draft/responseDraft.ts [hunk 2]
```diff
@@ -141,11 +153,18 @@ export class ResponseDraft extends DraftDocument {
...
   }
 
+  public isResponsePartiallyAdmittedWithInstalments (): boolean {
+    return this.isResponsePartiallyAdmitted()
+      && this.partialAdmission !== undefined
+      && this.partialAdmission.paymentOption !== undefined
+      && (this.partialAdmission.paymentOption.option === DefendantPaymentType.INSTALMENTS)
```
Comment (yarpo) [inline src/main/features/response/draft/responseDraft.ts]: why brackets only in this line?

File: src/main/features/response/helpers/statementOfMeansFeature.ts [hunk 3]
```diff
@@ -10,8 +10,10 @@ export class StatementOfMeansFeature {
     if (!draft) {
       throw new Error('Response draft is required')
     }
-    return draft.isResponseFullyAdmitted()
-      && (draft.isResponseFullyAdmittedWithPayBySetDate() || draft.isResponseFullyAdmittedWithInstalments())
+    const fullAdmissionHasStatementOfMeans = draft.isResponseFullyAdmittedWithPayBySetDate() || draft.isResponseFullyAdmittedWithInstalments()
+    const parAdmissionHasStatementOfMeans = draft.isResponsePartiallyAdmittedWithPayBySetDate() || draft.isResponsePartiallyAdmittedWithInstalments()
+    return ((draft.isResponseFullyAdmitted() && fullAdmissionHasStatementOfMeans)
+        || (draft.isResponsePartiallyAdmitted() && parAdmissionHasStatementOfMeans))
       && !draft.defendantDetails.partyDetails.isBusiness()
```
Comment (yarpo) [inline src/main/features/response/helpers/statementOfMeansFeature.ts]: can you please reformat it. Indentation `&& ...` seems to be incorrect.

File: src/main/features/response/views/confirmation/part-admission-steps.njk [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{% set claimantName = claim.claimData.claimant.name %}
+{% set amount = claim.response.paymentDetails.amount| numeral %}
+{% set fullAmount = claim.totalAmountTillToday| numeral %}
+
+{% if claim.response['isAlreadyPaid'] === 'yes' %}
```
Comment (yarpo) [inline src/main/features/response/views/confirmation/part-admission-steps.njk]: can you please use `YesNoOption.Yes.option`?

File: src/main/features/response/views/confirmation/part-admission-steps.njk [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{% set claimantName = claim.claimData.claimant.name %}
+{% set amount = claim.response.paymentDetails.amount| numeral %}
```
Comment (yarpo) [inline src/main/features/response/views/confirmation/part-admission-steps.njk]: can you please reformat it:
```
{% set amount = claim.response.paymentDetails.amount | numeral %}
```

File: src/main/features/response/views/confirmation/part-admission-steps.njk [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+{% set fullAmount = claim.totalAmountTillToday| numeral %}
```
Comment (yarpo) [inline src/main/features/response/views/confirmation/part-admission-steps.njk]: as above

File: src/main/features/response/views/confirmation/part-admission.njk [hunk 7]
```diff
@@ -0,0 +1,25 @@
+{% set claimantName = claim.claimData.claimant.name %}
+{% set claimantDetailsPageURI = ClaimPaths.claimantDetailsPage.evaluateUri({ externalId: claim.externalId }) %}
+{% set amount = claim.response.paymentDetails.amount| numeral %}
+
+{% if claim.response['isAlreadyPaid'] === 'yes' %}
```
Comment (yarpo) [inline src/main/features/response/views/confirmation/part-admission.njk]: `YesNoOption` please

File: src/main/features/response/views/confirmation/part-admission.njk [hunk 8]
```diff
@@ -0,0 +1,25 @@
+{% set claimantName = claim.claimData.claimant.name %}
+{% set claimantDetailsPageURI = ClaimPaths.claimantDetailsPage.evaluateUri({ externalId: claim.externalId }) %}
+{% set amount = claim.response.paymentDetails.amount| numeral %}
```
Comment (yarpo) [inline src/main/features/response/views/confirmation/part-admission.njk]: space before pipeline

File: src/main/features/response/views/confirmation/part-admission.njk [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+{% if claim.response['isAlreadyPaid'] === 'yes' %}
+  <p> {{ t('You told us you’ve paid the {{ amount }} you believe you owe. We’ve sent {{ claimantName }} this response.', { claimantName: claimantName, amount: amount }) }}</p>
+{% endif %}
+
+{% if claim.response.paymentOption %}
+  {% if claim.response.paymentDate %}
+    {% set paymentDate = claim.response.paymentDate | date %}
+  {% endif %}
+  {% if claim.response.paymentOption === domain.PaymentOption.IMMEDIATELY %}
+    {% set paymentMethod = 'immediately' %}
```
Comment (yarpo) [inline src/main/features/response/views/confirmation/part-admission.njk]: Isn't it `domain.PaymentOption.IMMEDIATELY.displayValue`?

File: src/main/features/response/views/confirmation/part-admission.njk [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  {% elseif claim.response.paymentOption === domain.PaymentOption.BY_SPECIFIED_DATE %}
+    {% set paymentMethod = 'by '+ paymentDate %}
```
Comment (yarpo) [inline src/main/features/response/views/confirmation/part-admission.njk]: Space before `+`

```
'by ' + paymentDate
```

File: src/main/app/claims/responseModelConverter.ts [hunk 11]
```diff
@@ -233,6 +267,17 @@ export class ResponseModelConverter {
     }
   }
 
+  private static convertPartPaymentDate (partAdmission: PartialAdmission): Moment {
```
Comment (sabahirfan) [inline src/main/app/claims/responseModelConverter.ts]: We have exactly same method in `responseModelConverter.ts`, please change the parameter and reuse same

File: src/main/features/response/views/confirmation/part-admission.njk (hunk 9 above)
Comment (sabahirfan) [reply src/main/features/response/views/confirmation/part-admission.njk]: We don't have display value in this object (`src/main/app/claims/models/response/core/paymentOption.ts`), so can't do that

**Pull request merged**